src/microbe_stage/editor/MicrobeEditor.cs

```diff
@@ -636,26 +636,27 @@ public void RemoveOrganelle()
     public float CalculateSpeed()
     {
         float finalSpeed = 0;
-        int flagCount = 0;
-        float lengthMicrobe = 0;
+        float massMicrobe = Constants.MICROBE_BASE_MASS;
+
+        float baseMovementForce = 0;
+        float organelleMovementForce = 0;
 
         foreach (var organelle in editedMicrobeOrganelles.Organelles)
         {
-            lengthMicrobe += organelle.Definition.HexCount;
+            massMicrobe += organelle.Definition.Mass;
 
-            // TODO: this should be changed to instead check for the
-            // movement component presence
-            if (organelle.Definition.InternalName == "flagellum")
-                flagCount++;
+            if (organelle.Definition.Components.Movement != null)
```

Comment:
This check should be:
`organelle.Definition.HasComponentFactory<MovementComponentFactory>()`